firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ApiClient.java

```diff
@@ -89,28 +89,35 @@ static FetchEligibleCampaignsResponse createCacheExpiringResponse() {
     }
     Logging.logi(FETCHING_CAMPAIGN_MESSAGE);
     providerInstaller.install();
-    return firebaseInstanceId
-        .getInstanceId()
-        .continueWith(
-            instanceIdResultTask -> {
-              InstanceIdResult instanceIdResult = instanceIdResultTask.getResult();
-              if (instanceIdResult == null) {
-                Logging.logw("InstanceID is null, not calling backend");
-                return createCacheExpiringResponse();
-              }
-              return withCacheExpirationSafeguards(
-                  grpcClient
-                      .get()
-                      .fetchEligibleCampaigns(
-                          FetchEligibleCampaignsRequest.newBuilder()
-                              // The project Id we expect is the gcm sender id
-                              .setProjectNumber(firebaseApp.getOptions().getGcmSenderId())
-                              .addAllAlreadySeenCampaigns(
-                                  impressionList.getAlreadySeenCampaignsList())
-                              .setClientSignals(getClientSignals())
-                              .setRequestingClientApp(getClientAppInfo(instanceIdResult))
-                              .build()));
-            });
+    return firebaseInstallations
+        .getId()
+        .continueWithTask(
```

Comment:
Why are you using continueWithTask? do you want to proceed even if this task is cancelled?

Reply:
so now I changed it to continueWith and I check both results against null. Am I not handling error cases correctly?

Reply:
I am actually not sure that null can be returned at all from those tasks, if an error happened `getResult()` will throw that error so I think it should be fine in terms of error handling.